cartoframes/data/enrichment/points_enrichment.py

```diff
@@ -0,0 +1,95 @@
+import uuid
+
+from . import bigquery_client
+from ...auth import get_default_credentials
+from collections import defaultdict
+
+
+_ENRICHMENT_ID = 'enrichment_id'
+
+# TODO: process column name in metadata, remove spaces and points
+
+
+def enrich_points(data, variables, data_geom_column='geometry', filters=dict()):
+
+    credentials = get_default_credentials()
+
+    bq_client = bigquery_client.BigQueryClient(credentials)
+
+    # Copy dataframe and generate id to join to original data later
+    data_copy = data.copy()
```

Comment:
I have been thinking about this. We can operate against the original dataframe provided by the user and then revert these operations in order to let the data provided by the user in the same state but what happens if the function is stopped in the middle of the operations, the revert operations can not be done so the data provided by will be modified.

Reply:
mmm, could we do kind of a view of the original dataframe to avoid the copy? I mean, if we don't do the `copy`, but just select both columns we need `data_copy[[data_geom_column, _ENRICHMENT_ID]]` + fill the enrichment ID, to a new dataframe,

I'd say this new dataframe is also a `reference` to the original one, but you didn't do a full copy (not sure about this though)

Not a big deal by now, but just telling to consider.